src/Engines/AlgoliaEngine.php

```diff
@@ -51,44 +51,56 @@ public function update($models)
             return;
         }
 
-        $index = $this->algolia->initIndex($models->first()->searchableAs());
+        $groupedByIndex = $models->groupBy(function ($model) {
+            return $model->searchableAs();
+        });
 
-        if ($this->usesSoftDelete($models->first()) && $this->softDelete) {
-            $models->each->pushSoftDeleteMetadata();
-        }
+        foreach ($groupedByIndex as $index => $collection) {
+            $index = $this->algolia->initIndex($index);
 
-        $objects = $models->map(function ($model) {
-            if (empty($searchableData = $model->toSearchableArray())) {
-                return;
+            if ($this->usesSoftDelete($collection->first()) && $this->softDelete) {
+                $collection->each->pushSoftDeleteMetadata();
             }
 
-            return array_merge(
-                ['objectID' => $model->getScoutKey()],
-                $searchableData,
-                $model->scoutMetadata()
-            );
-        })->filter()->values()->all();
+            $objects = $collection->map(function ($model) {
+                if (empty($searchableData = $model->toSearchableArray())) {
+                    return;
+                }
 
-        if (! empty($objects)) {
-            $index->saveObjects($objects);
+                return array_merge(
+                    ['objectID' => $model->getScoutKey()],
+                    $searchableData,
+                    $model->scoutMetadata()
+                );
+            })->filter()->values()->all();
+
+            if (!empty($objects)) {
+                $index->saveObjects($objects);
+            }
         }
     }
 
     /**
      * Remove the given model from the index.
      *
-     * @param  \Illuminate\Database\Eloquent\Collection  $models
+     * @param  \Illuminate\Database\Eloquent\Collection $models
```

Comment:
I guess this should be reverted too. The spaces come from the code style which is used for docblocks by laravel.